homeassistant/components/bluetooth/passive_update_coordinator.py

```diff
@@ -152,10 +109,121 @@ def _async_stop(self) -> None:
             self._cancel_track_unavailable()
             self._cancel_track_unavailable = None
 
+    @callback
+    def async_register_processor(
+        self, processor: PassiveBluetoothDataProcessor
+    ) -> Callable[[], None]:
+        """Register a processor that subscribes to updates."""
+        processor.coordinator = self
+
+        @callback
+        def remove_processor() -> None:
+            """Remove a processor."""
+            self._processors.remove(processor)
+            self._async_handle_processors_changed()
+
+        self._processors.append(processor)
+        self._async_handle_processors_changed()
+        return remove_processor
+
+    @callback
+    def _async_handle_processors_changed(self) -> None:
```

Comment:
Should we have some mechanism to stop or start the coordinator if all consumers (entities) are removed from all processors of the coordinator?

Reply:
We still need to get the updates in case there are new entities to be created.

Reply:
Right. Good.